flytekit/common/tasks/raw_container.py

```diff
@@ -0,0 +1,255 @@
+        cpu_limit: str = None,
+        gpu_limit: str = None,
+        memory_limit: str = None,
+        environment: Dict[str, str] = None,
+) -> _task_models.Container:
+    storage_limit = storage_limit or _resource_config.DEFAULT_STORAGE_LIMIT.get()
+    storage_request = storage_request or _resource_config.DEFAULT_STORAGE_REQUEST.get()
+    cpu_limit = cpu_limit or _resource_config.DEFAULT_CPU_LIMIT.get()
+    cpu_request = cpu_request or _resource_config.DEFAULT_CPU_REQUEST.get()
+    gpu_limit = gpu_limit or _resource_config.DEFAULT_GPU_LIMIT.get()
+    gpu_request = gpu_request or _resource_config.DEFAULT_GPU_REQUEST.get()
+    memory_limit = memory_limit or _resource_config.DEFAULT_MEMORY_LIMIT.get()
+    memory_request = memory_request or _resource_config.DEFAULT_MEMORY_REQUEST.get()
+
+    requests = []
+    if storage_request:
+        requests.append(
+            _task_models.Resources.ResourceEntry(
+                _task_models.Resources.ResourceName.STORAGE,
+                storage_request
+            )
+        )
+    if cpu_request:
+        requests.append(
+            _task_models.Resources.ResourceEntry(
+                _task_models.Resources.ResourceName.CPU,
+                cpu_request
+            )
+        )
+    if gpu_request:
+        requests.append(
+            _task_models.Resources.ResourceEntry(
+                _task_models.Resources.ResourceName.GPU,
+                gpu_request
+            )
+        )
+    if memory_request:
+        requests.append(
+            _task_models.Resources.ResourceEntry(
+                _task_models.Resources.ResourceName.MEMORY,
+                memory_request
+            )
+        )
+
+    limits = []
+    if storage_limit:
+        limits.append(
+            _task_models.Resources.ResourceEntry(
+                _task_models.Resources.ResourceName.STORAGE,
+                storage_limit
+            )
+        )
+    if cpu_limit:
+        limits.append(
+            _task_models.Resources.ResourceEntry(
+                _task_models.Resources.ResourceName.CPU,
+                cpu_limit
+            )
+        )
+    if gpu_limit:
+        limits.append(
+            _task_models.Resources.ResourceEntry(
+                _task_models.Resources.ResourceName.GPU,
+                gpu_limit
+            )
+        )
+    if memory_limit:
+        limits.append(
+            _task_models.Resources.ResourceEntry(
+                _task_models.Resources.ResourceName.MEMORY,
+                memory_limit
+            )
+        )
+
+    if environment is None:
+        environment = {}
+
+    return _task_models.Container(
+        image=image,
+        command=command,
+        args=args,
+        resources=_task_models.Resources(limits=limits, requests=requests),
+        env=environment,
+        config={}
+    )
+
+
+class SdkRawContainerTask(_base_task.SdkTask):
+    """
+    This class includes the logic for building a task that executes as a Presto task.
```

Comment:
Can we add a few paragraphs of documentation here or somewhere else on 1) how to use this and 2) how it works?

For 1) I've left you a page to fill in here: https://github.com/lyft/flytesnacks/blob/cookbook/cookbook/workflows/recipe_5/index.rst

Feel free to check out that branch and commit... or make your own PR if you want the street cred.

Reply:
nice :D will do